peps/pep-9999.rst

```diff
@@ -0,0 +1,820 @@
+   The rest of these guidelines don't apply to it.
+
+*  In **unstable** API (prefixed with ``PyUnstable_``), it is OK to ignore
+   these guidelines if there is a reason to do so -- usually for performance.
+   Please document the reason in a source comment.
+
+*  In the **general** public API, these guidelines apply in full force.
+   The C API working group can grant exceptions.
+
+*  For the **limited** API, please get explicit approval from the
+   C API working group for any changes.
+
+
+One header
+==========
+
+.. ref. https://github.com/capi-workgroup/api-evolution/issues/34
+
+All public API should be available after including ``Python.h``.
+
+To allow selecting an alternate API, such as a subset,
+use feature flags that users define before including the header
+(e.g. ``Py_LIMITED_API``).
+
+
+.. XXX add a PEP number prefix to the anchor:
+
+.. _naming:
+
+Naming
+======
+
+.. ref. https://github.com/capi-workgroup/api-evolution/issues/21
+
+All public names must be prefixed with ``Py``.
+
+Names that users should not use directly, but need to be visible to the
+compiler/linker, should be prefixed with ``_Py``.
+This is not considered public API.
+
+This applies to all names in a global namespace: functions, macros, variables,
+typedefs, structs, enums, etc.; not to parameters or struct fields.
+
+The ``Py_`` prefix is reserved for global service routines like
+``Py_FatalError``; specific groups of APIs use a longer prefix,
+e.g. ``PyUnicode_`` for string functions.
+Use an existing prefix when applicable. If you want to add a new prefix,
+contact the C API working group.
+
+The prefix is in mixed case, even in macro names (e.g. ``PyUnicode_AS_STRING``).
+
+Unstable API is prefixed with ``PyUnstable_`` instead of ``Py``,
+e.g. ``PyUnstable_Long_IsCompact`` or (hypothetically)
+``PyUnicode_String_GET_SIZE``.
+
+For common API concepts, use terms from :ref:`glossary`.
+Avoid synonyms, and avoid using glossary names for unrelated concepts.
+
+
+Do not reuse names
+------------------
+
+If a function's signature changes, or its behavior changes in
+a backwards-incompatible way, add a new function with a new name.
+You can deprecate and remove the old version following Python's
+:pep:`backwards compatibility policy <387>`.
+
+Even after API has been removed, do not reuse its old name,
+since existing documentation and tutorials will continue to refer to the
+old behavior.
+
+
+Language support
+================
+
+C standard and dialect
+----------------------
+
+.. ref. https://github.com/capi-workgroup/api-evolution/issues/22
+
+Public API must be compatible with:
+
+- C11, with optional features needed by CPython:
+
+  - IEEE 754 floating point (``__STDC_IEC_559__``)
+  - Atomics (``!__STDC_NO_ATOMICS__``, or MSVC)
+
+- C99
+- C88 with several select C99 features:
+
+  - ``<stdint.h>`` and ``<inttypes.h>``
+  - ``static inline`` functions
+  - designated initializers
+  - intermingled declarations
+  - line comments (``//``)
+
+- C++
```

Comment:
Which version of C++?

Reply:
Unspecified for now. See https://github.com/capi-workgroup/api-evolution/issues/22